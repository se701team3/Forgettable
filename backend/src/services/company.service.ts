/* eslint-disable no-underscore-dangle */
import mongoose from 'mongoose';
import Company, { CompanyModel } from '../models/company.model';
import logger from '../utils/logger';

const queryKeys = ['name', 'location', 'description'];

const createCompany = async (companyDetails: CompanyModel) => {
  const company = new Company(companyDetails);

  await company.save();

  return company;
};

const getCompany = async (companyId) => Company.findOne({ _id: companyId });

const getAllCompanies = async (queryParams: any, userCompanies: mongoose.Types.ObjectId[]) => {
  // Get all companies from the db which belongs to the user
  let foundUserCompanies = await Company.find({ _id: { $in: userCompanies } });

  // Filter the found companies by query params
  if (queryParams.term) {
    logger.info('Query params received:');
    logger.info(queryParams);
    const termValue = queryParams.term.toLowerCase();

    // If no relevant fields in an company match 'termValue', remove them from the array
    foundUserCompanies = foundUserCompanies.filter((company) => {
      for (let i = 0; i < queryKeys.length; i++) {
        // Make sure person has a value for current queryKey
        if (company[queryKeys[i]]) {
          const companyValue = (company[queryKeys[i]] as string).toLowerCase();
          if (companyValue.includes(termValue)) {
            return true;
          }
        }
      }
      return false;
    });
  }

  return foundUserCompanies;
};

const updateCompany = async (objectID: string, companyDetails: CompanyModel) => {
  console.log(objectID);
  const updatedCompany = await Company
    .findByIdAndUpdate(objectID, companyDetails, { new: true });

  return updatedCompany;
};

// Service is used for deletePersons endpoint
const deleteCompanyPerson = async (personID: string) => {
  // Remove the person with the supplied ID from all existing companies
  const result = await Company.updateMany({ }, { $pullAll: { persons: [{ _id: personID }] } });
  // Store references to the companys that no longer contain persons so they can be removed from Users later
  const empty_companies = await Company.find({ persons: { $exists: true, $size: 0 } });
  // Delete all the empty companies
  const deleted_companies = await Company.deleteMany({ persons: { $exists: true, $size: 0 } });

  // Check that all deleted companies are returned and companies have been updated
  if (empty_companies.length == deleted_companies.deletedCount) {
    return { array: empty_companies, bool: true };
  }
  return { array: empty_companies, bool: false };
};

const deleteCompany = async (companyID: String) => {
  const result = await Company.deleteOne({ _id: companyID }).exec();

  // Check that company has been deleted
  if (result.deletedCount == 1) {
    return true;
  }
  return false;
};

const companyService = {
  createCompany,
  updateCompany,
  getCompany,
  getAllCompanies,
  deleteCompany,
  deleteCompanyPerson,
};

export default companyService;
