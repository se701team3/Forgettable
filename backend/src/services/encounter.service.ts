import Encounter, {EncounterModel} from '../models/encounter.model';

export const createEncounter = async (encounterDetails: EncounterModel) => {
    const encounter = new Encounter(encounterDetails);
    await encounter.save();
    return encounter;
};

const encounterService = {
    createEncounter
  }

export default encounterService;