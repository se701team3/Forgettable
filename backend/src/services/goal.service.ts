import Goal, { GoalModel } from '../models/goal.model';

const createGoal = async (goalDetails: GoalModel) => {
  const goal = new Goal(goalDetails);

  // Uses the duration string and the current date to compute the start/end date of the goal
  goal.date_start = new Date(Date.now());
  goal.date_end = new Date();
  goal.date_end.setDate(goal.date_start.getDate() + parseInt(goal.duration));
  goal.encounter_goal = goalDetails.encounter_goal;

  await goal.save();

  if (goal.duration === null || goal.encounter_goal === null || goal.encounter_goal === 0 || goal.recurring === null) {
    await Goal.deleteOne({ _id: goal._id }).exec();
    const e = new Error('Goal fields can\'t be empty');
    e.name = 'ValidationError';
    throw e;
  }

  return goal;
};

const getGoal = async (goalId) => Goal.findOne({ _id: goalId });

const updateGoal = async (objectID: string, goalDetails: GoalModel) => {
  console.log(objectID);
  const updatedGoal = await Goal
    .findByIdAndUpdate(objectID, goalDetails, { new: true });

  return updatedGoal;
};

const deleteGoal = async (goalId: String) => {
  const result = await Goal.deleteOne({ _id: goalId }).exec();

  // Check that Goal has been deleted
  if (result.deletedCount == 1) {
    return true;
  }
  return false;
};

const goalService = {
  createGoal,
  getGoal,
  updateGoal,
  deleteGoal,
};

export default goalService;
