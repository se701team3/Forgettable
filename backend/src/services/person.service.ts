/**
 * Service contains database operations
 */
import mongoose from 'mongoose';
import Person, { PersonModel } from '../models/person.model';
import logger from '../utils/logger';

const queryKeys = ['first_name', 'last_name', 'gender', 'location', 'how_we_met', 'organisation'];

const createPerson = async (personDetails: PersonModel) => {
  const person = new Person(personDetails);
  await person.save();
  return person;
};

const updatePersonWithId = async (reqPersonId: string, personNewDetails: PersonModel) => {
  const query = { _id: reqPersonId };
  return Person.findOneAndUpdate(query, personNewDetails, { upsert: true });
};

const getPersonWithId = async (reqPersonId: string) => {
  const query = { _id: reqPersonId };
  return Person.findOne(query);
};

const getPeople = async (queryParams: any, userPersons: mongoose.Types.ObjectId[]) => {
  // Get all persons from the db that belong to the user
  let foundUserPersons = await Person.find({ _id: { $in: userPersons } });

  // Filter the found persons by query params
  if (queryParams.term) {
    logger.info('Query params received:');
    logger.info(queryParams);
    const termValue = queryParams.term.toLowerCase();

    // If no relevant fields in a Person match 'termValue', remove them from the array
    foundUserPersons = foundUserPersons.filter((person) => {
      for (let i = 0; i < queryKeys.length; i++) {
        // Make sure person has a value for current queryKey
        if (person[queryKeys[i]]) {
          const personValue = (person[queryKeys[i]] as string).toLowerCase();
          if (personValue.includes(termValue)) {
            return true;
          }
        }
      }
      return false;
    })
  }

  return foundUserPersons;
};

const deletePersonEncounters = async (encounterID: string) => {
  const result = await Person.updateMany({ }, { $pullAll: {encounters: [{ _id: encounterID}]} });

  // Check that Persons with the respective encounters has been updated
  if (result.modifiedCount > 0) {
    return true;
  } else {
    return false;
  }
}

const deletePersons = async (personID: string) => {
  await Person.deleteOne({_id: personID});
}

const addEncounterToPersons = async (personIds, encounterId) => {
  for (let i = 0; i < personIds.length; i++) {
    const result = await Person
      .updateOne({ _id: personIds[i] }, { $push: { encounters: encounterId } });

    // Revert all updates if any update fails
    if (result.modifiedCount !== 1) {
      for (let j = i - 1; j >= 0; j--) {
        await Person.updateOne({ _id: personIds[i] }, { $pop: { encounters: 1 } });
      }
      return false;
    }
  }

  return true;
};

const personService = {
  createPerson,
  getPersonWithId,
  getPeople,
  deletePersonEncounters,
  deletePersons,
  addEncounterToPersons,
  updatePersonWithId,
};

export default personService;
