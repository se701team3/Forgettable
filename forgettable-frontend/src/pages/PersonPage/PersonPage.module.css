.PersonPage {
    width: 100%;
    height: 90%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    overflow-x: none;
}

.ContentContainer {
    width: 100%;
    height: 90%;
    box-sizing: border-box;
    padding: 0 60px;
    overflow-y: none;
}

.TitleContainer {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
    align-items: flex-end;
    margin-top: 80px;
    margin-bottom: 50px;
    width: 805px;
}

.TitleContainer > h1 {
    font-size: 50px;
    font-weight: 600;
    line-height: 75px;
    margin: 0;
}

.ButtonContainer {
    margin-top: 20px;
}

.EncountersContainer {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    flex-wrap: wrap;
    width: 100%;
}

.CardWrapper {
    margin: 0 40px 40px 0;
}

.DeleteModal {
    width: 400px;
    display: flex;
    flex-direction: column ;
    align-items: center;
}

.DeleteModal > h1 {
    color: #E05A33;
    font-size: var(--text-xxlarge);
    text-align: center;
    font-weight: 600;
}

.DeleteModal > p {
    text-align: center;
    font-size: var(--text-large);
}