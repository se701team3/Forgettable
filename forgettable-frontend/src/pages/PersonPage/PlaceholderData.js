export const ENCOUNTERS = [
  {
    title: 'Aenean et quam',
    description: 'Nunc aliquet leo vitae sapien varius ullamcorper. Donec eget vestibulum turpis. Suspendisse dolor urna, ultricies in neque quis, dictum sodales arcu. ',
    persons: [
      {
        first_name: 'Mercury',
        last_name: 'Lin',
        img: 'https://user-images.githubusercontent.com/62003343/158300159-495244dd-f05e-4384-9fac-ddf88a06be39.gif',
      },
    ],
    location: 'Auckland',
    date: new Date(),
  },
  {
    title: 'Sed a est venenatis',
    description: 'Sed sed urna dui. Nam gravida, leo ac molestie pulvinar, odio erat egestas lorem, in ullamcorper lacus eros eu purus. Proin nulla neque, tristique ac euismod ut, convallis vitae mi. ',
    persons: [
      {
        first_name: 'Mercury',
        last_name: 'Lin',
        img: 'https://user-images.githubusercontent.com/62003343/158300159-495244dd-f05e-4384-9fac-ddf88a06be39.gif',
      },
      {
        first_name: 'Raina',
        last_name: 'Song',
        img: 'https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg',
      },
      {
        first_name: 'Hazel',
        last_name: 'Williams',
        img: 'https://www.publicdomainpictures.net/pictures/320000/nahled/background-image.png',
      },
    ],
    location: 'Auckland',
  },
  {
    title: 'Good times!',
    description: 'Diam dictum vestibulum mi nulla vestibulum, id nibh. Nunc consequat amet commodo turpis tellus. Scelerisque a pellentesque vel accumsan sed mauris, ac turpis pharetra. Sem tristique nulla cursus praesent tincidunt integer faucibus in aliquam. Pretium nam enim ut suspendisse dictum. Arcu tristique turpis nam nunc nisl. Mi, molestie quis tincidunt ipsum hendrerit. Lacinia at ut dui nibh placerat scelerisque congue egestas ut. Et fusce odio amet, etiam consectetur ipsum a. Adipiscing iaculis dictum nisl lobortis. Pharetra nunc semper adipiscing massa aenean eu iaculis risus. At quam feugiat lorem ac dui. A, facilisi lectus mus iaculis convallis proin semper dolor accumsan. Ut nibh quis porttitor faucibus sagittis netus sit quis.',
    persons: [
      {
        first_name: 'Mercury',
        last_name: 'Lin',
        img: 'https://user-images.githubusercontent.com/62003343/158300159-495244dd-f05e-4384-9fac-ddf88a06be39.gif',

      }],
    location: 'Auckland',
  },
  {
    title: 'Good times!',
    description: 'Diam dictum vestibulum mi nulla vestibulum, id nibh. Nunc consequat amet commodo turpis tellus. Scelerisque a pellentesque vel accumsan sed mauris, ac turpis pharetra. Sem tristique nulla cursus praesent tincidunt integer ...',
    persons: [
      {
        first_name: 'Mercury',
        last_name: 'Lin',
        img: 'https://user-images.githubusercontent.com/62003343/158300159-495244dd-f05e-4384-9fac-ddf88a06be39.gif',

      },
    ],
    location: 'Auckland',
    date: new Date(),
  },
];
