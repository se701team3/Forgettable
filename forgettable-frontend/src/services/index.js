import {getData, postData, putData, deleteData} from './functions/helpers';

/**
 * Fetches data of a single Person, given the Person's id
 * @param {string} id id of the person to retrieve
 * @return {Promise} data of the person with the id
 */
export const getPerson = async (id) => {
  return await getData('persons/' + id);
};

/**
 * Fetches data of a single Company, given the Company's id
 * @param {string} id id of the Company to retrieve
 * @return {Promise} data of the company with the id
 */
export const getCompany = async (id) => {
  return await getData('company/' + id);
};

/**
 * Fetches data of all Persons
 * @return {Promise} data of all persons
 */
export const getAllPersons = async (page, limit) => {
  if (!page || !limit) return await getData('persons');
  return await getData(`persons?page=${page}&limit=${limit}`);
};

/**
 * Fetches data of all Companies
 * @return {Promise} data of all companies
 */
export const getAllCompanies = async (page, limit) => {
  if (!page || !limit) return await getData('company');
  return await getData(`company?page=${page}&limit=${limit}`);
};

/**
 * Fetches data of the user given their id
 * @return {Promise} data of the user using the system
 */
export const getUser = async () => {
  return await getData('users');
};

/**
 * Creates a User in the database given the User's data
 * @param {Object} user data of the user to create.
 * Must include a first_name property.
 * @return {Promise} data entry of the new Person created
 */
export const createUser = async (user) => {
  return await postData('users', user);
};

/**
 * Creates a Person in the database given the Person's data
 * @param {Object} person data of the person to create.
 * Must include a first_name property.
 * @return {Promise} data entry of the new Person created
 */
export const createPerson = async (person) => {
  return await postData('persons', person);
};

/**
 * Creates a Company in the database given the Company's data
 * @param {Object} company data of the Company to create.
 * Must include a name property.
 * @return {Promise} data entry of the new Company created
 */
export const createCompany = async (company) => {
  return await postData('company', company);
};

/**
 * Updates a Person in the database given the Person's new data.
 * New data will overwrite all old data.
 * @param {string} id id of the Person to update
 * @param {Object} person data of the Person to update.
 * @return {Promise} data entry of the updated Person
 */
export const updatePerson = async (id, person) => {
  return await putData('persons/' + id, person);
};

/**
 * Updates a Company in the database given the Company's new data.
 * New data will overwrite all old data.
 * @param {string} id id of the Company to update
 * @param {Object} company data of the Company to update.
 * @return {Promise} data entry of the updated Company
 */
export const updateCompany = async (id, company) => {
  return await putData('company/' + id, company);
};

/**
 * Deletes a Person from the database given the Person's id
 * @param {string} id id of the Person to delete
 * @return {Promise}
 */
export const deletePerson = async (id) => {
  return await deleteData('persons/' + id);
};

/**
 * Deletes a Company from the database given the Company's id
 * @param {string} id id of the Company to delete
 * @return {Promise}
 */
export const deleteCompany = async (id) => {
  return await deleteData('company/' + id);
};

/**
 * Fetches the data of a single Encounter, given the Encounter's id
 * @param {string} id id of the Encounter to fetch
 * @return {Promise} data of the Encounter with the id
 */
export const getEncounter = async (id) => {
  return await getData('encounters/' + id);
};

/**
 * Fetches all Encounters that the user has been part of
 * @return {Promise} data of all Encounters of the user
 */
export const getAllEncounters = async () => {
  return await getData('encounters');
};

/**
 * Fetches Encounters entries with pagination
 * @param {string} page starting page (starting index = 1)
 * @param {string} limit how many entries on one page
 * @return {Promise} data of all Encounters of the user
 */
export const getEncountersByPage = async (page, limit) => {
  if (!page || !limit) return await getAllEncounters();
  return await getData(`encounters/?page=${page}&limit=${limit}`);
};

/**
 * Creates a new Encounter in the database given the Encounter's data
 * @param {Object} encounter data of the Encounter to create.
 * @return {Promise} data entry of the new Encounter created
 */
export const createEncounter = async (encounter) => {
  return await postData('encounters', encounter);
};

/**
 * Updates an Encounter in the database given the Encounter's new data.
 * New data will overwrite all old data.
 * @param {string} id id of the Encounter to update
 * @param {Object} encounter data to update the Encounter with.
 * @return {Promise} data entry of the updated Encounter
 */
export const updateEncounter = async (id, encounter) => {
  return await putData('encounters/' + id, encounter);
};

/**
 * Deletes an Encounter from the database given the Encounter's id
 * @param {string} id id of the Encounter to delete
 * @return {Promise}
 */
export const deleteEncounter = async (id) => {
  return await deleteData('encounters/' + id);
};

/**
 * Searches Encounters from the database given a search string
 * @param {string} searchString string to search
 * @param {string} field the field that is going to be search
 * @return {Promise}
 */
export const searchEncounter = async (searchString, field) => {
  return await getData('encounters/?term=' + searchString + '&field=' + field);
};

/**
 * Searches Persons from the database given a search string
 * @param {string} searchString string to search
 * @param {string} field the field that is going to be search
 * @return {Promise}
 */
export const searchPersons = async (searchString, field) => {
  return await getData('persons/?term=' + searchString + '&field=' + field);
};

/**
 * Searches persons with upcoming birthdays on your list
 * @return {Promise}
 */
export const getPeopleWithUpcomingBirthday = async () => {
  return await getData('birthdays');
};

/**
 * Searches Company from the database given a search string
 * @param {string} searchString string to search
 * @return {Promise}
 */
export const searchCompany = async (string) => {
  return await getData('company/?term=' + string);
};
