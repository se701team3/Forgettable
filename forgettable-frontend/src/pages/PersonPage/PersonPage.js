import React, {useEffect, useState} from 'react';
import EncounterCard from '../../components/EncounterCard/EncounterCard';
import IconButton from '../../components/IconButton/IconButton';
import PersonDrawer from '../../components/PersonDrawer/PersonDrawer';
import {createPerson, getPerson} from '../../services';
import classes from './PersonPage.module.css';
import {ENCOUNTERS} from './PlaceholderData';
import {useParams} from 'react-router-dom';

const PersonPage = (props) => {
  // TODO: this needs to be passed in as a routing parameter
  const {id} = useParams();

  const [person, setPerson] = useState({
    firstName: '',
    lastName: '',
    birthday: null,
    gender: '',
    location: '',
    howWeMet: '',
    firstMet: null,
    interests: [],
    organisation: '',
    socialMedia: [],
    img: '',
    encounters: [],
    timeUpdated: null,
  });

  useEffect(async () => {
    const result = await getPerson(id);

    setPerson({
      firstName: result.first_name,
      lastName: result.last_name,
      birthday: result.birthday,
      gender: result.gender,
      location: result.location,
      howWeMet: result.how_we_met,
      interests: result.interests,
      organisation: result.organisation,
      socialMedia: convertSocialMedia(result.socialMedia),
      img: result.image,
      encounters: result.encounters,
      encounters: ENCOUNTERS,
      timeUpdated: result.timeUpdated,
    });
  }, [id]);

  const convertSocialMedia = (socialMedias) => {
    if (!socialMedias) return null;

    const socialMediaArray = [];

    for (const [key, value] of Object.entries(socialMedias)) {
      socialMediaArray.push({
        name: key,
        link: value,
      });
    }

    return socialMediaArray;
  };

  const createEncounterComponent = (encounter, i) => {
    return (
      <div className={classes.CardWrapper} key={encounter.id || i}>
        <EncounterCard
          title={encounter.title}
          description={encounter.description}
          persons={encounter.persons}
          location={encounter.location}
          onClick={() => {}}
          onDelete={() => {}}
          date={encounter.date}
          isInitialEncounter={false}
        />
      </div>
    );
  };

  return (
    <div className={classes.PersonPage}>
      <PersonDrawer
        open={true}
        img={person.img}
        staticDrawer={true}
        name={`${person.firstName} ${person.lastName}`}
        firstMet={person.firstMet}
        interests={person.interests}
        organisation={person.organisation}
        location={person.location}
        gender={person.gender}
        birthday={person.birthday}
        socialMedias={person.socialMedia}
        data-testid="drawer-component"
      />
      <div className={classes.ContentContainer}>
        <div className={classes.TitleContainer} >
          <h1 className={classes.Title}>
            You encountered
            <br/>
            {person.firstName} {person.encounters.length} times
          </h1>
          <div className={classes.ButtonContainer}>
            <IconButton
              btnText="New Encounter"
              onClick={() => {}}
              includeIcon={true}
              height="66px"
            />
          </div>
        </div>
        <div className={classes.EncountersContainer}>
          {
            person.encounters.map((encounter, i) => {
              return createEncounterComponent(encounter, i);
            })
          }
        </div>
      </div>
    </div>
  );
};

export default PersonPage;
