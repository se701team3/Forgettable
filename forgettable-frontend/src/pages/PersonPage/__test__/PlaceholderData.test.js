import React from 'react';
import {screen, render, fireEvent, waitFor} from '@testing-library/react';
import {act} from 'react-dom/test-utils';
import * as api from '../../../services';
import ShallowRenderer from 'react-test-renderer/shallow';

import PersonPage from '../PersonPage';

jest.mock('../../../services');

it('renders PersonPage UI with correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<PersonPage/>);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});


it('should display the correct user returned from the person api call', async () => {
  api.getPerson.mockImplementation(() =>
    Promise.resolve({
      first_name: 'Hiruna',
      last_name: 'Smith',
    }),
  );

  act(() => {
    render(<PersonPage />);
  });

  await waitFor(
      () => expect(screen.getByText('Hiruna Smith'))
          .toBeInTheDocument());
});

it('should display the correct title detailing the number of encounters returned from person api call', async () => {
  api.getPerson.mockImplementation(() =>
    Promise.resolve({
      first_name: 'Hiruna',
      last_name: 'Smith',
      encounters: [{
        title: 'encounter 1',
        description: 'description 1',
        persons: [
          {
            first_name: 'person 1',
            last_name: 'person 1',
          },
        ],
        location: 'place 1',
        onClick: jest.fn(),
        onDelete: jest.fn(),
        isInitialEncounter: false,
      }, {
        title: 'encounter 2',
        description: 'description 2',
        persons: [
          {
            first_name: 'person 2',
            last_name: 'person 2',
          },
        ],
        location: 'place 2',
        onClick: jest.fn(),
        onDelete: jest.fn(),
        isInitialEncounter: false,
      },
      ],
    }),
  );

  act(() => {
    render(<PersonPage />);
  });

  await waitFor(
      () => expect(screen.getByText('2 times'))
          .toBeInTheDocument());
});
