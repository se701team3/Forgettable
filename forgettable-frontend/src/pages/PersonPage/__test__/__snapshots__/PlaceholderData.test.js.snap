// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PersonPage UI with correct hierarchy 1`] = `
<div
  className="PersonPage"
>
  <PersonDrawer
    birthday={null}
    data-testid="drawer-component"
    firstMet={null}
    gender=""
    img=""
    interests={Array []}
    location=""
    name=" "
    open={true}
    organisation=""
    socialMedias={Array []}
    staticDrawer={true}
  />
  <div
    className="ContentContainer"
  >
    <div
      className="TitleContainer"
    >
      <h1
        className="Title"
      >
        You encountered
        <br />
        
         
        0
         times
      </h1>
      <div
        className="ButtonContainer"
      >
        <IconButton
          btnText="New Encounter"
          height="66px"
          includeIcon={true}
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="EncountersContainer"
    />
  </div>
</div>
`;
